TAScheduler/views.py

```diff
@@ -178,6 +178,77 @@ def post(request):
         # TODO: implement post
         pass
 
+class DeleteLab(View):
+    def get(request):
+        # If the user does not have a valid name, I.E. if they try to manually enter /home in the search bar,
+        # they will fail the userAllowed test and be redirected back to the login page
+        # If the user is allowed then home is rendered like normal
+        if userAllowed(request, ["SUPERVISOR"]):
+            return render(request, "deletelab.html", {"lab_list":Lab.objects.all()})
+        else:
+            return redirect("/../home/")
+
+    def post(request):
+        delete = True
+        try:
+            delete_or_submit = int(request.POST["delete"])
+        except MultiValueDictKeyError:
+            delete_or_submit = int(request.POST["submit"])
+            delete = False
+        if delete:
+            deleted_lab = Lab.objects.get(labID = delete_or_submit)
+            Lab.objects.delete(labID=delete_or_submit)
+            return render(request, "deletelab.html", {"lab_list":Lab.objects.all(), "deleted_lab": deleted_lab})
+        else:
+            return render(request, "deletelab.html", {"lab_list": Lab.objects.all()})
```

Comment:
```suggestion
        LabManagement.deleteLab(lab_id=request.POST("delete")
        return render(request, "delete.html", {"lab_list": Lab.objects.all())
```
Since we don't need to actually process any input apart from the id of the lab needing to be deleted, this code can be simplified greatly.